bin/brew

```diff
@@ -68,12 +68,61 @@ then
   then
     HOMEBREW_PREFIX="/usr/local"
   fi
+  unset USR_LOCAL_BREW_FILE_DIRECTORY USR_LOCAL_HOMEBREW_REPOSITORY
 fi
 
+unset BREW_FILE_DIRECTORY
+
 # If the location of HOMEBREW_LIBRARY changes
 # keg_relocate.rb, formula_cellar_checks.rb, and test/global_spec.rb need to change.
 HOMEBREW_LIBRARY="${HOMEBREW_REPOSITORY}/Library"
 
+# Load Homebrew's variable configuration files from disk.
+# First, load the system-wide configuration.
+if [[ -f "/etc/homebrew/brew.env" ]]
+then
+  unset SYSTEM_ENV_TAKES_PRIORITY
+  # only load HOMEBREW_*=* lines
+  SYSTEM_HOMEBREW_ENV="$(grep -E '^HOMEBREW_[A-Z_]+=.*$' "/etc/homebrew/brew.env")"
+  if [[ -n "${HOMEBREW_SYSTEM_ENV_TAKES_PRIORITY-}" ]]
+  then
+    SYSTEM_ENV_TAKES_PRIORITY="1"
+  else
+    eval "${SYSTEM_HOMEBREW_ENV}"
+  fi
+fi
+
+# Next, load the prefix configuration
+if [[ -f "${HOMEBREW_PREFIX}/etc/homebrew/brew.env" ]]
+then
+  # only load HOMEBREW_*=* lines
+  PREFIX_HOMEBREW_ENV="$(grep -E '^HOMEBREW_[A-Z_]+=.*$' "${HOMEBREW_PREFIX}/etc/homebrew/brew.env")"
+  eval "${PREFIX_HOMEBREW_ENV}"
+  unset PREFIX_HOMEBREW_ENV
+fi
+
+# Finally, load the user configuration
+if [[ -n "${XDG_CONFIG_HOME-}" ]]
+then
+  HOMEBREW_USER_CONFIG_HOME="${XDG_CONFIG_HOME}/homebrew"
```

Comment:
```suggestion
if [[ -n "${HOMEBREW_XDG_CONFIG_HOME-}" ]]
then
  HOMEBREW_USER_CONFIG_HOME="${HOMEBREW_XDG_CONFIG_HOME}/homebrew"
```
and `XDG_CONFIG_HOME` needs to be added to `USED_BY_HOMEBREW_VARS` in `bin/brew`.

Reply:
@carlocab Can you explain that? I don't think either should be needed.

Reply:
Env filtering:
```
❯ XDG_CONFIG_HOME=foo brew ruby -e 'puts ENV["XDG_CONFIG_HOME"].present?'
false
```

Reply:
@carlocab We don't rely on `XDG_CONFIG_HOME` being set anywhere after env filtering is applied (unless I missed something), no?

Reply:
@carlocab that behaviour is already present. We don't pass through `XDG_CONFIG_HOME` because we don't want/need to. It's documented that we use it if set but that's only needed/used in `brew.sh` which in turn sets `HOMEBREW_USER_CONFIG_HOME` accordingly and that is used afterwards.

For an example:
```console
$ export XDG_CONFIG_HOME=fish[~/OSS/Homebrew]
$ brew irb
==> Interactive Homebrew Shell
...
brew(main):001:0> ENV["HOMEBREW_USER_CONFIG_HOME"]
=> "fish/homebrew"
```